src/UniversalRewardsDistributor.sol

```diff
@@ -17,238 +18,164 @@ import {MerkleProof} from "@openzeppelin/utils/cryptography/MerkleProof.sol";
     }
 
-    modifier onlyTreasury(uint256 distributionId) {
-        require(msg.sender == treasuryOf[distributionId], ErrorsLib.CALLER_NOT_TREASURY);
+    modifier onlyOwner() {
+        require(msg.sender == owner, ErrorsLib.CALLER_NOT_OWNER);
         _;
     }
 
-    modifier onlyOwner(uint256 distributionId) {
-        require(msg.sender == ownerOf[distributionId], ErrorsLib.CALLER_NOT_OWNER);
-        _;
+    /// @notice Initializes the contract.
+    /// @param initialOwner The initial owner of the contract.
+    /// @param initialTimelock The initial timelock of the contract.
+    /// @param initialRoot The initial merkle tree's root.
+    /// @param initialIpfsHash The optional ipfs hash containing metadata about the root (e.g. the merkle tree itself).
+    constructor(address initialOwner, uint256 initialTimelock, bytes32 initialRoot, bytes32 initialIpfsHash) {
+        owner = initialOwner;
+        timelock = initialTimelock;
+
+        if (initialRoot != bytes32(0)) {
+            _forceUpdateRoot(initialRoot, initialIpfsHash);
+        }
     }
 
     /* EXTERNAL */
 
     /// @notice Proposes a new merkle tree root.
-    /// @param distributionId The distributionId of the merkle tree distribution.
     /// @param newRoot The new merkle tree's root.
-    /// @param ipfsHash The optional ipfs hash containing metadata about the root (e.g. the merkle tree itself).
-    function proposeRoot(uint256 distributionId, bytes32 newRoot, bytes32 ipfsHash)
-        external
-        onlyUpdater(distributionId)
-    {
-        if (timelockOf[distributionId] == 0) {
-            _forceUpdateRoot(distributionId, newRoot, ipfsHash);
+    /// @param newIpfsHash The optional ipfs hash containing metadata about the root (e.g. the merkle tree itself).
+    function proposeRoot(bytes32 newRoot, bytes32 newIpfsHash) external onlyUpdater {
+        if (timelock == 0) {
+            _forceUpdateRoot(newRoot, newIpfsHash);
         } else {
-            pendingRootOf[distributionId] = PendingRoot(block.timestamp, newRoot, ipfsHash);
-            emit RootProposed(distributionId, newRoot, ipfsHash);
+            pendingRoot = PendingRoot(block.timestamp, newRoot, newIpfsHash);
+            emit EventsLib.RootProposed(newRoot, newIpfsHash);
         }
     }
 
     /// @notice Accepts the current pending merkle tree's root.
-    /// @param distributionId The distributionId of the merkle tree distribution.
     /// @dev This function can only be called after the timelock has expired.
     /// @dev Anyone can call this function.
-    function acceptRootUpdate(uint256 distributionId) external {
-        PendingRoot memory pendingRoot = pendingRootOf[distributionId];
-        require(pendingRoot.submittedAt > 0, ErrorsLib.NO_PENDING_ROOT);
-        require(block.timestamp >= pendingRoot.submittedAt + timelockOf[distributionId], ErrorsLib.TIMELOCK_NOT_EXPIRED);
+    function acceptRootUpdate() external {
+        PendingRoot memory pendingRootMem = pendingRoot;
```

Comment:
The compiler will optimize for this natively (and may be more performant if `PendingRoot`  is longer than 32 bytes) so I would remove such memory assignments from storage